From 4c745aae959b04b16abd1559ded6bd11617d46be Mon Sep 17 00:00:00 2001
From: Ivan Korniienko <ivan.o.korniienko@globallogic.com>
Date: Wed, 27 Nov 2019 13:23:56 +0200
Subject: [PATCH] Make SecureElementService visible for instant apps

SecureElementService should be visible for instant apps to be able to
pass instant test cases on aosp_car_x86_64 emulator.

Bug: 145094907
Test: emulator -wipe-data # or "adb reboot" on real car device
      cts-tradefed run cts -m CtsSecureElementAccessControlTestCases1
Jira-Id: ADM-3975
Signed-off-by: Ivan Korniienko <ivan.o.korniienko@globallogic.com>
Change-Id: Ib187c8555ef60d6f138f6b7e7eb6a4d474a9941c
(cherry picked from commit 99dc3c9e50d34275179be0aa85e562320c72bc96)
---
 AndroidManifest.xml | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 3412d66..c4006df 100755
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -4,8 +4,10 @@
           android:sharedUserId="android.uid.se">
     <application android:name=".SEApplication"
                  android:label="SecureElementApplication"
-                 android:persistent="true">
-        <service android:name=".SecureElementService">
+                 android:persistent="true"
+                 android:directBootAware="true">
+        <service android:name=".SecureElementService"
+                 android:visibleToInstantApps="true">
             <intent-filter>
                 <action android:name="android.omapi.ISecureElementService"/>
             </intent-filter>
-- 
2.28.0

